lib/instrumentation/index.js

```diff
@@ -76,11 +103,20 @@ Instrumentation.prototype.start = function () {
     require('./patch-async')(this)
   }
 
+  this._startHook()
+}
+
+Instrumentation.prototype._startHook = function () {
+  if (this._hook) {
+    this._hook.unhook()
+  }
+
+  var self = this
   var disabled = new Set(this._agent._conf.disableInstrumentations)
 
   this._agent.logger.debug('adding hook to Node.js module loader')
 
-  this._hook = hook(MODULES, function (exports, name, basedir) {
+  this._hook = hook(this._patches.keys, function (exports, name, basedir) {
```

Comment:
Where is this `keys` property coming from?

Reply:
fixed